0006-cdt-protocol/config-updated.avsc

```diff
@@ -1,13 +1,63 @@
-{"namespace": "org.kaaproject.ipc.events.gen.v1",
- "type": "record",
- "name": "ConfigUpdated",
- "fields": [
-    {"name": "correlationId", "type": "string", "default":""},
-    {"name": "timestamp", "type": "long", "default":-1},
-    {"name": "timeout", "type": "long", "default":-1},
-    {"name": "originatorReplicaId", "type": "string", "default":""},
-    {"name": "appVersionName", "type": "string", "default": ""},
-    {"name": "endpointId", "type": "string", "default": ""},
-    {"name": "configId", "type": "string", "default":""}
- ]
+{
+    "namespace":"org.kaaproject.ipc.event.gen.v1.endpoint.config",
+    "type":"record",
+    "name":"ConfigUpdated",
+    "doc":"Broadcast message to indicate that endpoint configuration data got updated.",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "default":"",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":-1,
+            "doc":"The amount of milliseconds since the timestamp until the message expires. Value of -1 is reserved to indicate no expiration."
+        },
+        {
+            "name":"appVersionName",
+            "type":"string",
+            "default":"",
+            "doc":"The application version name, for which the configuration was updated"
+        },
+        {
+            "name":"endpointId",
+            "type":"string",
+            "default":"",
+            "doc":"Unique endpoint identifier, for which the configuration was updated"
+        },
+        {
+            "name":"configId",
+            "type":"string",
+            "default":"",
+            "doc":"The unique identifier of the new configuration."
```

Comment:
We had a similar discussion re. JIRA ticket subjects. The outcome: use a period when it's grammatically accurate: at the end of a complete sentence that is a statement.

> A simple sentence contains a single subject and predicate. It describes only one thing, idea or question, and has only one verb - it contains only an independent (main) clause. An independent clause contains a subject and verb and expresses a complete thought in both context and meaning.

Thus, in most cases when doc does not contain a complete sentence, you don't need a period.
> The unique ID

You may need it, though, to separate several statements:
> A unique ID. Optional.

I'll do an extra pass to normalize this in this RFC.